### DISCUSSION TITLE
Plaintext Credentials saved in Citect SCADA code files

### DISCUSSION THREAD

**Discussion by JakeSmithQUT** [cutaway-security/IACS_STAR_Methodology#11] [Use Cases]
# Title: Plaintext Credentials saved in Citect SCADA code files

# Scope
The Citect SCADA project uses its proprietary programming language Cicode to execute custom functions. These Cicode functions are called by any Citect process (IO Server, Trend Server, Client [UI], Reporting Server) to monitor and control plant equipment and data as well as perform common user actions programmatically.

# Finding
Site used the Cicode function _Login(username, password)_ to programmatically login users on SCADA software startup. The use of this function meant the username and password credentials of these SCADA accounts were stored in plaintext in the Cicode files. Some of the accounts discovered in this vulnerability had elevated permissions. Testing showed that these credentials could be extracted with a non-elevated powershell script using REGEX to identify the Login function within all files of the *.ci type within the project directories. These account credentials were present in both the SCADA project backup files and the deployed systems. Each copy of the project (on both operator terminals, backup files, server installations, etc...) contained the same Cicode files meaning that all of the exposed credentials were present in every site SCADA instance.

# Countermeasures

- Complete removal of the Login command from Cicode files within the Citect project. 
- Changing of exposed account credentials. 
- Encryption of scheduled SCADA backups. 
- Monitoring of Cicode files for Login function use.

# Impact / Consequences
Users with access to these accounts are able to make control systems changes to all PLC devices by toggling and modifying tag values having the ability to cause significant production disruption. These accounts can be accessed from any Citect SCADA Terminal located physically on site and are able to issue control commands and modify setpoints. As a result, if exploited, an adversary would be able to achieve 'Loss of Availability' and 'Manipulation of Control ' though the toggling of commands and the changing of setpoints to force equipment to stop or trip, naturally also leading to 'Loss of Productivity and Revenue'.

# Factor Selections
  * Likelihood Factor: Medium (LF: 4.125)
  * Consequence Factor: Low (CF: 2.625)
  * Overall Risk Severity: Low
  * Score Vector: [(SL:0/M:0/O:0/A:0/EA:0/EE:0/AW:0/DR:0/LC:0/LI:0/LA:0/LAC:0/ED:0/PD:0/SE:0/R:0)](https://iacs-star-calculator.com/?vector=(SL:0/M:0/O:0/A:0/EA:0/EE:0/AW:0/DR:0/LC:0/LI:0/LA:0/LAC:0/ED:0/PD:0/SE:0/R:0))

**Comment by cutaway** [comment]
@JakeSmithQUT Your Score Vector did not copy and paste correctly. Can you provide an update? (FYI, we are working on fixing the vector score feature.)

Thank you, cutaway
